fail2ban/client/fail2bancmdline.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/python
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: t -*-
+# vi: set ft=python sts=4 ts=4 sw=4 noet :
+#
+# This file is part of Fail2Ban.
+#
+# Fail2Ban is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# Fail2Ban is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Fail2Ban; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+__author__ = "Fail2Ban Developers"
+__copyright__ = "Copyright (c) 2004-2008 Cyril Jaquier, 2012-2014 Yaroslav Halchenko, 2014-2016 Serg G. Brester"
+__license__ = "GPL"
+
+import getopt
+import logging
+import os
+import sys
+
+from ..version import version
+from ..protocol import printFormatted
+from ..helpers import getLogger
+
+# Gets the instance of the logger.
+logSys = getLogger("fail2ban")
+
+def output(s): # pragma: no cover
+	print(s)
+
+CONFIG_PARAMS = ("socket", "pidfile", "logtarget", "loglevel", "syslogsocket",)
+# Used to signal - we are in test cases (ex: prevents change logging params, log capturing, etc)
+PRODUCTION = True
+
+MAX_WAITTIME = 30
+
+class Fail2banCmdLine():
+
+	def __init__(self):
+		self._argv = self._args = None
+		self._configurator = None
+		self.resetConf()
+
+	def resetConf(self):
+		self._conf = {
+		  "async": False,
+			"conf": "/etc/fail2ban",
+			"force": False,
+			"background": True,
+			"verbose": 1,
+			"socket": None,
+			"pidfile": None,
+			"timeout": MAX_WAITTIME
+		}
+
+	@property
+	def configurator(self):
+		if self._configurator:
+			return self._configurator
+		# New configurator
+		from .configurator import Configurator
+		self._configurator = Configurator()
+		# Set the configuration path
+		self._configurator.setBaseDir(self._conf["conf"])
+		return self._configurator
+
+
+	def applyMembers(self, obj):
+		for o in obj.__dict__:
+			self.__dict__[o] = obj.__dict__[o]
+
+	def dispVersion(self):
+		output("Fail2Ban v" + version)
+		output("")
+		output("Copyright (c) 2004-2008 Cyril Jaquier, 2008- Fail2Ban Contributors")
+		output("Copyright of modifications held by their respective authors.")
+		output("Licensed under the GNU General Public License v2 (GPL).")
+		output("")
+		output("Written by Cyril Jaquier <cyril.jaquier@fail2ban.org>.")
+		output("Many contributions by Yaroslav O. Halchenko <debian@onerussian.com>.")
```

Comment:
within this PR or some time later, we should remove those 2 lines here and in places alike.  Fail2Ban now is far beyond Cyril+mine's work.  Generic copyright statement should be sufficient I guess